### PR TITLE
fix support for high sierra

### PR DESCRIPTION
**Pull request by kristenlc**
downloading gaster fails on high sierra, it must be included by the repo or the script will not work
it is not good to download always the latest gaster using the script anyway, as what if palera1n is no longer maintained but then there are lots of updates to gaster years down the line, then the script will no longer work cause it is using years newer gaster version instead of the version the script was designed for. it is always advisable to include the version that the script is known to work with.

img4tool and gtar needs to be compiled on high sierra to work on high sierra otherwise it compiles with symbols that are only present in newer macos versions and the script fails miserably 

also set -e needs to be commented out in palera1n.sh as otherwise the script just executes and prints nothing for some reason

btw i tested your AOP fixes and it did not fix camera or deep sleep bug on a11, but i did verify that your script with these changes work perfectly on high sierra and i am able to boot the ramdisk, dump blobs, boot palera1n with amfi patches on ios 15.4.1 on iphone x, and open tips app and hit install to install sileo, and sileo works great.

i am creating this pr not with the expectation that you will accept the pr, but more as guidance so you know what needs to be done generally to get palera1n working on high sierra for your users, as i am going to be archiving my repo and ending all support for my version of palera1n. i will update my readme and youtube video to point everyone to your repo. thank you for all that you do.

### CODE REVIEW

File: palera1n.sh [hunk 1]
```diff
@@ -4,7 +4,7 @@ mkdir -p logs
 
 {
 
-set -e
+#set -e
```
Comment (itsnebulalol) [inline palera1n.sh:7]: Will the error handler still work?

Comment (kristenlc) [reply palera1n.sh:7]: yes

**Comment by itsnebulalol** [comment]
Does gtar and img4tool work on modern macOS, I can test later if you’re not sure

**Comment by kristenlc** [comment]
> Does gtar and img4tool work on modern macOS, I can test later if you’re not sure

It should because it is a self contained static binary, no dependencies . I have not tested though .
You may add a check and only run set -e if the os is higher then high sierra, as it does interfere with your error handler.

**Comment by iam-theKid** [comment]
installing libgeneral should be enough to include a compiled binary of img4tool for High Sierra;
gtar does not have dependencies as far as I know;

Then including a compiled binary for High Sierra and another for newer versions of img4tool and gtar should do the trick, in the code, i have something like this

if [[ "$(sw_vers | grep 'ProductVersion' | cut -d ':' -f2 | cut -d ' ' -f2)" =~ 10.13.* ]]; then
img4toolV="img4tool10.13"
gtarV="gtar10.13"
else
img4tolV="img4tool"
gtarV="gtar"
fi

img4toolV="img4tool10.13"

**Comment by RavanGami** [comment]
image

**Comment by RavanGami** [comment]
image

**Comment by kristenlc** [comment]
> image

This exact error shown in this screenshot is a very well known issue on older version of macos, use [SSHRD_Script-High-Sierra](https://github.com/kristenlc/SSHRD_Script-High-Sierra/tree/main) and follow the instructions on that page. This version I am linking you to is unsupported though, so if it does not work out for you then you will have to open your own issue in the original repository.

**Pull request closed without merging**